File: Models/Entities/Product.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;


public class Product
{

    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    public required string productId { get; set; }

    public required string productName { get; set; }

    public required string productType { get; set; }

    public string vendorId { get; set; }

    public int? quantity { get; set; }

    public bool isActive {get; set;}

    public int? lowerMargin { get; set; }

    public bool? restockRequired { get; set; } = false;

    public string? productDescription { get; set; }

    [BsonDateTimeOptions(Kind = DateTimeKind.Local)]
    public DateTime createdDate { get; set; }

    [BsonDateTimeOptions(Kind = DateTimeKind.Local)]
    public DateTime updatedDate { get; set; }

    //image
    public byte[]? productImage { get; set; }

    

}

File: android/app/src/main/java/com/example/markethub/domain/models/Product.kt
package com.example.markethub.domain.models

import com.example.markethub.BuildConfig

data class Product(
    val productId: String,
    val productName: String,
    val productType: String,
    val productDescription: String? = null,
    val productImage: String? = null,
    val quantity: Int? = null,
    val price: Double,
    val rating: Rating,
    val isActive: Boolean,
    val vendor: VendorDetails? = null
){
    val fullImageUrl: String?
        get() = productImage?.let { "${BuildConfig.BASE_URL}$it" }
}

data class VendorDetails(
    val vendorId: String,
    val vendorName: String,
    val rating: Rating
)
